services/housetables/src/main/java/com/linkedin/openhouse/housetables/services/UserTablesServiceImpl.java

```diff
@@ -46,12 +48,41 @@ public UserTableDto getUserTable(String databaseId, String tableId) {
 
   @Override
   public List<UserTableDto> getAllUserTables(UserTable userTable) {
-    UserTableDto targetUserTableDto = userTablesMapper.fromUserTable(userTable);
-
-    return StreamSupport.stream(htsJdbcRepository.findAll().spliterator(), false)
-        .map(userTableRow -> userTablesMapper.toUserTableDto(userTableRow))
-        .filter(x -> x.match(targetUserTableDto))
-        .collect(Collectors.toList());
+    if (isListDatabases(userTable)) {
+      // list databases
+      return StreamSupport.stream(
+              htsJdbcRepository.findAllDistinctDatabaseIds().spliterator(), false)
+          .map(databaseId -> UserTableDto.builder().databaseId(databaseId).build())
+          .collect(Collectors.toList());
+    } else if (isListTables(userTable)) {
+      // list tables in a database
+      return StreamSupport.stream(
+              htsJdbcRepository
+                  .findAllByDatabaseIdIgnoreCase(userTable.getDatabaseId())
+                  .spliterator(),
+              false)
+          .map(userTableRow -> userTablesMapper.toUserTableDto(userTableRow))
+          .collect(Collectors.toList());
+    } else if (isListTablesWithPattern(userTable)) {
+      // list tables with tableId or with pattern in a database
+      return StreamSupport.stream(
+              htsJdbcRepository
+                  .findAllByDatabaseIdAndTableIdLikeAllIgnoreCase(
+                      userTable.getDatabaseId(), userTable.getTableId())
+                  .spliterator(),
+              false)
+          .map(userTableRow -> userTablesMapper.toUserTableDto(userTableRow))
+          .collect(Collectors.toList());
+    } else {
+      // general search
+      log.warn(
```

Comment:
This should never happen since validation layer kind of covers this? also if these calls indeed happen due to a miss how are we planning to track them if we actually have handling? checking warn messages might not scale?

Reply:
Added metrics so we can set up alerts later.